component/main.jsonnet

```
@@ -0,0 +1,43 @@
+// main template for minio
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.minio;
+
+local namespace = kube.Namespace(params.namespace) {
+  metadata+: {
+    labels+: {
+      app: 'minio',
+      'app.kubernetes.io/name': 'minio',
+      'app.kubernetes.io/instance': 'minio',
+    },
+  },
+};
+
+local accessKey = kube.Secret(params.accessKey) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  // need to use stringData here for secret reveal to work
+  stringData+: {
+    secretkey: params.accessKey,
+  },
+};
+
+local secretKey = kube.Secret(params.secretKey) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  // need to use stringData here for secret reveal to work
+  stringData+: {
+    secretkey: params.accessKey,
+  },
+};
```

```suggestion
local adminCredentials = kube.Secret(params.adminCredentials.secretName) {
  metadata+: {
    namespace: params.namespace,
  },
  // need to use stringData here for secret reveal to work
  // use keys which bitnami/minio Helm chart expects for the credentials secret
  stringData+: {
    access-key: params.adminCredentials.accessKey,
    secret-key: params.adminCredentials.secretKey,
  },
};
```

I'd suggest creating the secret here in the format which the bitnami/minio Helmchart expects. That way, we can simply provide the name of our secret as `existingSecret` when rendering the Helm chart to actually configure the Minio instance to use the referenced access/secret key in Vault.